Description: Run the autostart file for Unity, too.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No, Unity/Ubuntu specific

Index: gnome-screensaver-3.5.5/src/gnome-screensaver.desktop.in
===================================================================
--- gnome-screensaver-3.5.5.orig/src/gnome-screensaver.desktop.in	2012-08-07 09:02:58.000000000 +0200
+++ gnome-screensaver-3.5.5/src/gnome-screensaver.desktop.in	2012-08-07 09:05:13.775635591 +0200
@@ -4,7 +4,7 @@
 _Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity
 AutostartCondition=GNOME3 if-session gnome-fallback
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
Index: gnome-screensaver-3.5.5/src/gnome-screensaver.desktop.in.in
===================================================================
--- gnome-screensaver-3.5.5.orig/src/gnome-screensaver.desktop.in.in	2012-08-07 09:02:58.000000000 +0200
+++ gnome-screensaver-3.5.5/src/gnome-screensaver.desktop.in.in	2012-08-07 09:05:07.603635808 +0200
@@ -4,7 +4,7 @@
 _Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity
 AutostartCondition=GNOME3 if-session gnome-fallback
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
