--- a/usr/lib/linuxmint/mintMenu/plugins/execute.py
+++ b/usr/lib/linuxmint/mintMenu/plugins/execute.py
@@ -32,7 +32,7 @@ def Execute( cmd , commandCwd=None):
 			except Exception, detail:
 				print detail
 				return False
-		cmd = cmd.split()
+	cmd = cmd.split()
 	cmd = RemoveArgs(cmd)
 	
 	try:

--- a/usr/lib/linuxmint/mintMenu/plugins/system_management.py
+++ b/usr/lib/linuxmint/mintMenu/plugins/system_management.py
@@ -140,11 +140,11 @@ class pluginclass( object ):
                 self.mintMenuWin.setTooltip( Button1, _("Browse and install available software") )
 
         if ( self.showPackageManager == True ):
-            Button2 = easyButton( "applications-system", self.iconsize, [_("Package Manager")], -1, -1 )
-            Button2.connect( "clicked", self.ButtonClicked, "gksu /usr/sbin/synaptic" )
+            Button2 = easyButton( "software-manager", self.iconsize, [_("Package Manager")], -1, -1 )
+            Button2.connect( "clicked", self.ButtonClicked, "pamac-manager" )
             Button2.show()
             self.systemBtnHolder.pack_start( Button2, False, False, 0 )
-            self.mintMenuWin.setTooltip( Button2, _("Install, remove and upgrade software packages") )
+            self.mintMenuWin.setTooltip( Button2, _("Install, remove, and update software packages") )
 
         if ( self.showControlCenter == True ):
             Button3 = easyButton( "gtk-preferences", self.iconsize, [_("Control Center")], -1, -1 )
@@ -178,7 +178,7 @@ class pluginclass( object ):
                 if os.path.exists("/usr/bin/mate-screensaver-command"):
                     Button5.connect( "clicked", self.ButtonClicked, "mate-screensaver-command -l" )
                 else:
-                    Button5.connect( "clicked", self.ButtonClicked, "xdg-screensaver lock" )
+                    Button5.connect( "clicked", self.ButtonClicked, "loginctl lock-sessions" )
                 Button5.show()
                 self.systemBtnHolder.pack_start( Button5, False, False, 0 )
                 self.mintMenuWin.setTooltip( Button5, _("Requires password to unlock") )
