--- a/usr/lib/linuxmint/mintMenu/mintMenuConfig.py
+++ b/usr/lib/linuxmint/mintMenu/mintMenuConfig.py
@@ -386,8 +386,11 @@ class mintMenuConfig( object ):
         self.headingColorLabel.set_sensitive(  widget.get_active() )

     def getBackgroundColor( self ):
-        color = Gdk.Color(0,0,0)
-        self.backgroundColor.get_color(color)
+        try:
+            color = self.backgroundColor.get_color()
+        except TypeError:
+            color = Gdk.Color(0, 0, 0)
+            self.backgroundColor.get_color(color)
         return self.gdkColorToString( color )

     def getBorderColor( self ):