diff -aur light-locker-settings-1.5.0.clean/light-locker-settings/light-locker-settings.py light-locker-settings-1.5.0/light-locker-settings/light-locker-settings.py
--- light-locker-settings-1.5.0.clean/light-locker-settings/light-locker-settings.py	2015-01-19 21:07:06.000000000 -0600
+++ light-locker-settings-1.5.0/light-locker-settings/light-locker-settings.py	2015-05-03 07:34:05.362267224 -0500
@@ -354,8 +354,11 @@
         for directory in (GLib.get_system_config_dirs()):
             dirs.append (os.path.join(directory, 'autostart'))
 
-        keyfile.load_from_dirs(filename, dirs,
-                               GLib.KeyFileFlags.KEEP_TRANSLATIONS)
+        try:
+            keyfile.load_from_dirs(filename, dirs,
+                                   GLib.KeyFileFlags.KEEP_TRANSLATIONS)
+        except Exception:
+            pass
 
         for key in defaults.keys():
             try:
