--- a/src/nemo-pathbar.c
+++ b/src/nemo-pathbar.c
@@ -616,7 +616,9 @@
     if (path_bar->priv->button_list == NULL) {
         return;
     }
-    direction = gtk_widget_get_direction (widget);
+    gtk_widget_get_preferred_width (path_bar->priv->up_slider_button,
+                       &path_bar->priv->slider_width,
+                       NULL);
 
     gtk_widget_get_preferred_size (BUTTON_DATA (path_bar->priv->button_list->data)->button,
                        NULL, &child_requisition);