diff -crB pantheon-greeter_original/data/org.pantheon.desktop.greeter.gschema.xml pantheon-greeter_new/data/org.pantheon.desktop.greeter.gschema.xml
*** pantheon-greeter_original/data/org.pantheon.desktop.greeter.gschema.xml	2013-01-29 22:27:08.221240000 +0100
--- pantheon-greeter_new/data/org.pantheon.desktop.greeter.gschema.xml	2013-01-29 22:55:15.127570151 +0100
***************
*** 26,32 ****
      </key>
  
      <key name="default-wallpaper" type="s">
!       <default><![CDATA['/usr/share/backgrounds/elementaryos-default']]></default>
        <summary>Default Wallpaper</summary>
        <description>Default wallpaper used for guest accounts and those which haven't set a wallpaper.</description>
      </key>
--- 26,32 ----
      </key>
  
      <key name="default-wallpaper" type="s">
!       <default><![CDATA['/usr/share/backgrounds/cinnarch-default']]></default>
        <summary>Default Wallpaper</summary>
        <description>Default wallpaper used for guest accounts and those which haven't set a wallpaper.</description>
      </key>
diff -crB pantheon-greeter_original/src/Indicators.vala pantheon-greeter_new/src/Indicators.vala
*** pantheon-greeter_original/src/Indicators.vala	2013-01-29 22:27:08.221240000 +0100
--- pantheon-greeter_new/src/Indicators.vala	2013-01-29 22:49:22.157542311 +0100
***************
*** 213,222 ****
          if (LightDM.get_can_suspend ())
              power.submenu.append (suspend);
  
!         if (LightDM.get_can_restart ())
              power.submenu.append (restart);
  
!         if (LightDM.get_can_shutdown ())
              power.submenu.append (poweroff);
  
          poweroff.activate.connect (() => {try { LightDM.shutdown (); } catch (Error e) { warning (e.message); }});
--- 213,222 ----
          if (LightDM.get_can_suspend ())
              power.submenu.append (suspend);
  
! 
              power.submenu.append (restart);
  
! 
              power.submenu.append (poweroff);
  
          poweroff.activate.connect (() => {try { LightDM.shutdown (); } catch (Error e) { warning (e.message); }});
diff -crB pantheon-greeter_original/src/PantheonGreeter.vala pantheon-greeter_new/src/PantheonGreeter.vala
*** pantheon-greeter_original/src/PantheonGreeter.vala	2013-01-29 22:27:08.221240000 +0100
--- pantheon-greeter_new/src/PantheonGreeter.vala	2013-01-29 22:52:50.328311664 +0100
***************
*** 296,304 ****
      Gdk.get_default_root_window ().set_cursor (new Gdk.Cursor (Gdk.CursorType.LEFT_PTR));
  
      var settings = Gtk.Settings.get_default ();
!     settings.gtk_theme_name = "elementary";
!     settings.gtk_icon_theme_name = "elementary";
!     settings.gtk_font_name = "Droid Sans";
      settings.gtk_xft_dpi= (int) (1024 * 96);
      settings.gtk_xft_antialias = 1;
      settings.gtk_xft_hintstyle = "hintslight";
--- 296,304 ----
      Gdk.get_default_root_window ().set_cursor (new Gdk.Cursor (Gdk.CursorType.LEFT_PTR));
  
      var settings = Gtk.Settings.get_default ();
!     settings.gtk_theme_name = "Adwaita";
!     settings.gtk_icon_theme_name = "Faenza-Dark";
!     settings.gtk_font_name = "Open Sans";
      settings.gtk_xft_dpi= (int) (1024 * 96);
      settings.gtk_xft_antialias = 1;
      settings.gtk_xft_hintstyle = "hintslight";
