diff -ruN a/in.json/Makefile b/in.json/Makefile
--- a/in.json/Makefile	2014-12-30 20:01:39.978997830 +0100
+++ b/in.json/Makefile	2014-12-30 20:08:45.569012561 +0100
@@ -1,7 +1,7 @@
 PREFIX  = /usr
 
-GOPATH     := $(shell go env GOPATH)
-QT5_LIBDIR ?= $(PREFIX)/lib/qt5
+GOPATH     := $(PREFIX)/lib/go
+QT5_LIBDIR ?= $(PREFIX)/lib/qt
 QMLDIR     ?= $(QT5_LIBDIR)/qml
 
 JSONS  := $(wildcard *.in.json)
@@ -55,9 +55,9 @@
 	@rm -rf go qml
 
 install-golang: build-golang
-	@echo copying go source files to $(DESTDIR)$(GOPATH)/src ...
-	@mkdir -p $(DESTDIR)$(GOPATH)/src
-	@cp -r go/src/dbus $(DESTDIR)$(GOPATH)/src/
+	@echo copying go source files to $(DESTDIR)$(GOPATH)/site/src ...
+	@mkdir -p $(DESTDIR)$(GOPATH)/site/src
+	@cp -r go/src/dbus $(DESTDIR)$(GOPATH)/site/src/
 
 install-qml: build-qml
 	@echo install qml plugins to $(DESTDIR)$(QMLDIR)
