diff -upr pluma-orig/configure.ac pluma/configure.ac
--- pluma-orig/configure.ac	2015-11-18 17:06:44.008107000 +0100
+++ pluma/configure.ac	2015-11-20 20:01:52.895098695 +0100
@@ -386,21 +386,9 @@ plugin-loaders/python/bindings/Makefile
 plugins/Makefile
 plugins/changecase/Makefile
 plugins/docinfo/Makefile
-plugins/externaltools/data/Makefile
-plugins/externaltools/Makefile
-plugins/externaltools/scripts/Makefile
-plugins/externaltools/tools/Makefile
 plugins/filebrowser/Makefile
 plugins/filebrowser/org.mate.pluma.plugins.filebrowser.gschema.xml
 plugins/modelines/Makefile
-plugins/pythonconsole/Makefile
-plugins/pythonconsole/pythonconsole/Makefile
-plugins/quickopen/Makefile
-plugins/quickopen/quickopen/Makefile
-plugins/snippets/data/lang/Makefile
-plugins/snippets/data/Makefile
-plugins/snippets/Makefile
-plugins/snippets/snippets/Makefile
 plugins/sort/Makefile
 plugins/spell/Makefile
 plugins/spell/org.mate.pluma.plugins.spell.gschema.xml
diff -upr pluma-orig/plugins/Makefile.am pluma/plugins/Makefile.am
--- pluma-orig/plugins/Makefile.am	2015-11-18 13:01:07.391541000 +0100
+++ pluma/plugins/Makefile.am	2015-11-20 20:01:56.818137767 +0100
@@ -1,12 +1,8 @@
 DIST_SUBDIRS =		\
 	changecase 	\
 	docinfo 	\
-	externaltools	\
 	filebrowser 	\
 	modelines	\
-	pythonconsole	\
-	quickopen	\
-	snippets	\
 	sort 		\
 	spell 		\
 	taglist 	\
@@ -16,7 +12,6 @@ DIST_SUBDIRS =		\
 SUBDIRS = 		\
 	changecase	\
 	docinfo		\
-	externaltools \
 	filebrowser	\
 	modelines	\
 	sort		\
@@ -24,10 +19,6 @@ SUBDIRS = 		\
 	time		\
 	trailsave
 
-if ENABLE_PYTHON
-SUBDIRS      += pythonconsole snippets quickopen
-endif
-
 if ENABLE_ENCHANT
 SUBDIRS      += spell
 endif
